### PR TITLE
Adding New k8s.io lists for api-machinery in order to fix Zoom accounts and prepare for future list migrations

### PR DESCRIPTION
**Pull request by fedebongio**
Adding new lists to SIG API Machinery to fix zoom account problems and prepare for future lists migration

### CODE REVIEW

File: groups/sig-api-machinery/groups.yaml [hunk 1]
```diff
@@ -7,6 +7,42 @@ groups:
...
+    owners:
+      - jpbetz@google.com
+      - deads@redhat.com
+      - fbongiovanni@google.com
+      - stefan.schimanski@gmail.com
+    settings:
+      AllowWebPosting: "true"
+      ReconcileMembers: "true"
+      WhoCanPostMessage: "ANYONE_CAN_POST"
+      WhoCanViewGroup: "ALL_MEMBERS_CAN_VIEW"
```
Comment (ameukam) [inline groups/sig-api-machinery/groups.yaml:26]:
```suggestion
- email-id: sig-api-machinery-leads@kubernetes.io
  name: sig-api-machinery-leads
  description: |-
    SIG api-machinery leads
  settings:
    WhoCanPostMessage: "ANYONE_CAN_POST"
    ReconcileMembers: "true"
  owners:
    - jpbetz@google.com
    - deads@redhat.com
    - fbongiovanni@google.com
    - stefan.schimanski@gmail.com
  settings:
    AllowWebPosting: "true"
    ReconcileMembers: "true"
    WhoCanPostMessage: "ANYONE_CAN_POST"
    WhoCanViewGroup: "ALL_MEMBERS_CAN_VIEW"
```

Comment (fedebongio) [reply groups/sig-api-machinery/groups.yaml:26]: Did I add unnecessary spaces?

File: groups/sig-api-machinery/groups.yaml [hunk 2]
```diff
@@ -7,40 +7,73 @@ groups:
...
-      - kermitalexandr@google.com
-      - leilajal@google.com
-      - smythe@google.com
-      - stevenlinde@google.com
-      - stclair@google.com
+  owners:
+     - jpbetz@google.com
+     - cic37@google.com
+     - deads@redhat.com
+     - lavalamp@google.com
```
Comment (logicalhan) [inline groups/sig-api-machinery/groups.yaml:54]: :(

Comment (fedebongio) [reply groups/sig-api-machinery/groups.yaml]: :(

**Comment by fedebongio** [comment]
/hold 

I'll squash the commits when I get a second

**Comment by fedebongio** [comment]
tide/merge-method-squash

**Pull request merged**